### ISSUE TITLE
Cryptic Error when fitting (but only occurs when script is NOT run on C:\Users\{USERNAME})

### ISSUE THREAD

**Issue by RJsing2011** [catboost/catboost#2570]
Problem: Cryptic error when running .fit()
catboost version: catboost==1.2.2
Operating System: Windows 10 Pro
CPU: Yes
GPU: No

I'm fitting a set of data to a binary outcome. When I run the script on the path where catboost dependencies are installed (i.e. C:\Users\{USERNAME}), no error occurs and the fitting runs fine. But when I run the script on a different path the error below occurs. Unfortunately, after "CatBoostError:" the message stops, so I don't have any clue what could be the problem. Any clue what this error message could mean?

File ~\anaconda3\lib\site-packages\catboost\core.py:5100, in CatBoostClassifier.fit(self, X, y, cat_features, text_features, embedding_features, sample_weight, baseline, use_best_model, eval_set, verbose, logging_level, plot, plot_file, column_description, verbose_eval, metric_period, silent, early_stopping_rounds, save_snapshot, snapshot_file, snapshot_interval, init_model, callbacks, log_cout, log_cerr)
   5097 if 'loss_function' in params:
   5098     CatBoostClassifier._check_is_compatible_loss(params['loss_function'])
-> 5100 self._fit(X, y, cat_features, text_features, embedding_features, None, sample_weight, None, None, None, None, baseline, use_best_model,
   5101           eval_set, verbose, logging_level, plot, plot_file, column_description, verbose_eval, metric_period,
   5102           silent, early_stopping_rounds, save_snapshot, snapshot_file, snapshot_interval, init_model, callbacks, log_cout, log_cerr)
   5103 return self

File ~\anaconda3\lib\site-packages\catboost\core.py:2319, in CatBoost._fit(self, X, y, cat_features, text_features, embedding_features, pairs, sample_weight, group_id, group_weight, subgroup_id, pairs_weight, baseline, use_best_model, eval_set, verbose, logging_level, plot, plot_file, column_description, verbose_eval, metric_period, silent, early_stopping_rounds, save_snapshot, snapshot_file, snapshot_interval, init_model, callbacks, log_cout, log_cerr)
   2315 allow_clear_pool = train_params["allow_clear_pool"]
   2317 with log_fixup(log_cout, log_cerr), \
   2318     plot_wrapper(plot, plot_file, 'Training plots', [_get_train_dir(self.get_params())]):
-> 2319     self._train(
   2320         train_pool,
   2321         train_params["eval_sets"],
   2322         params,
   2323         allow_clear_pool,
   2324         train_params["init_model"]
   2325     )
   2327 # Have property feature_importance possibly set
   2328 loss = self._object._get_loss_function_name()

File ~\anaconda3\lib\site-packages\catboost\core.py:1723, in _CatBoostBase._train(self, train_pool, test_pool, params, allow_clear_pool, init_model)
   1722 def _train(self, train_pool, test_pool, params, allow_clear_pool, init_model):
-> 1723     self._object._train(train_pool, test_pool, params, allow_clear_pool, init_model._object if init_model else None)
   1724     self._set_trained_model_attributes()

File _catboost.pyx:4645, in _catboost._CatBoost._train()

File _catboost.pyx:4694, in _catboost._CatBoost._train()

CatBoostError:

**Comment by RJsing2011** [comment]
Hello!
Many thanks for looking into this!

Unfortunately, setting train_dir didn't do the trick. It's the exact same error popping up again. I've also tried to set allow_writing_files=False and save_snapshot=False but didn't make no difference. Is there any other way to prevent catboost from writing files? Or any other idea what it could be? It's probably also worth mentioning that the error doesn't always pop at at the same iteration of the fitting process. It's rather random...

I'm really in despair... :(

**Comment by RJsing2011** [comment]
Nevermind, I did make it work, you were right with the train_dir! I first set the variable wrong. Thanks a lot! You really helped me a lot with this one! I appreciate it.

> Hello! Probably, there are no rights to write auxiliary files (for example, snapshots), and the specified `train_dir` option would help.
